### PR TITLE
Migrate `pallet-session` to the new pallet macro

### PR DESCRIPTION
**Pull request by koushiro**
Part of #7882

Migrate `pallet-session` (excluding historical) to the new pallet attribute macro.

## :warning:  Breaking Change :warning:

From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

> storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the pallet must be careful about the name they used in `construct_runtime!`. Hence the `runtime-migration` label, which might not be needed depending on the configuration of the pallet.

`polkadot` and `kusama` use `Session` as pallet name, thus no need for migration

### CODE REVIEW

**Review by emostov — APPROVED** [review]
lgtm

File: frame/session/src/historical/offchain.rs [hunk 1]
```diff
@@ -30,15 +30,11 @@ use sp_runtime::{
...
-use super::{
-	super::{Pallet as SessionModule, SessionIndex},
-	Config, IdentificationTuple, ProvingTrie,
-};
-
-use super::shared;
 use sp_std::prelude::*;
 
+use super::{shared, Config, IdentificationTuple, ProvingTrie};
+use crate::{Pallet as SessionModule, SessionIndex};
```
Comment (kianenigma) [inline frame/session/src/historical/offchain.rs:36]: thanks for clearing these up, `super::super` is super confusing (pun intended).

**Review by kianenigma — COMMENTED** [review]
LGTM

**Comment by KiChjang** [comment]
I wonder if it's okay to just leave `pallet-session-historical` to be using declarative macros instead of the attribute macros?

**Comment by koushiro** [comment]
> I wonder if it's okay to just leave `pallet-session-historical` to be using declarative macros instead of the attribute macros?

why use declarative macro？I will do it in another PR

**Comment by KiChjang** [comment]
I meant whether it's okay to _not_ change it in this PR and wait for the next one. But I suppose there isn't any harm in merging this as-is...

**Review by KiChjang — APPROVED** [review]

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**